src/qt/bitcoingui.cpp

```diff
@@ -1378,24 +1378,18 @@ void BitcoinGUI::showNormalIfMinimized(bool fToggleHidden)
     if(!clientModel)
         return;
 
-    // activateWindow() (sometimes) helps with keyboard focus on Windows
-    if (isHidden())
-    {
-        show();
-        activateWindow();
-    }
-    else if (isMinimized())
-    {
-        showNormal();
+    if (!isHidden() && !isMinimized() && !GUIUtil::isObscured(this) && fToggleHidden) {
+        hide();
+    } else {
+        // activateWindow() (sometimes) helps with keyboard focus on Windows
+        if (isMinimized()) {
+            showNormal();
+        } else {
+            show();
+        }
         activateWindow();
```

Comment:
I think you need to add also `raise()` here, to bring the window on top of others.

Reply:
Done.